lib/Doctrine/DBAL/Query/Expression/CompositeExpression.php

```diff
@@ -91,7 +91,7 @@ public function addMultiple(array $parts = [])
      */
     public function add($part)
     {
-        if ( ! empty($part) || ($part instanceof self && $part->count() > 0)) {
+        if ( ! empty($part) && ! ($part instanceof self && $part->count() === 0)) {
```

Comment:
Can you refactor this conditional to remove some negations? It is extremely hard to read.